lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureLocation.java

```diff
@@ -105,7 +106,7 @@ public String account()
 
     public String path()
     {
-        return location.path();
+        return Utility.urlEncode(location.path());
```

Comment:
what this detected by `TestDeltaLakeAdlsConnectorSmokeTest`
if so, the code fix should come before switching the test to use native fs

Reply:
It was only detected by the product tests, so the commits are in the right order now.